src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py

```diff
@@ -1076,6 +1077,96 @@ def internal_get_description(dbg, seq, thread_id, frame_id, expression):
         dbg.writer.add_command(cmd)
 
 
+def internal_get_exception_details_json(dbg, request, thread_id, set_additional_thread_info=None, iter_visible_frames_info=None):
+    ''' Fetch exception details
+    '''
+    try:
+        thread = pydevd_find_thread_by_id(thread_id)
+        additional_info = set_additional_thread_info(thread)
+        topmost_frame = additional_info.get_topmost_frame(thread)
+
+        frames = []
+        exc_type = None
+        exc_desc = None
+        if topmost_frame is not None:
+            frame_id_to_lineno = {}
+            try:
+                trace_obj = None
+                frame = topmost_frame
+                while frame is not None:
+                    if frame.f_code.co_name == 'do_wait_suspend' and frame.f_code.co_filename.endswith('pydevd.py'):
+                        arg = frame.f_locals.get('arg', None)
+                        if arg is not None:
+                            exc_type, exc_desc, trace_obj = arg
+                            break
+                    frame = frame.f_back
+
+                while trace_obj.tb_next is not None:
+                    trace_obj = trace_obj.tb_next
+
+                info = dbg.suspended_frames_manager.get_topmost_frame_and_frame_id_to_line(thread_id)
+                if info is not None:
+                    topmost_frame, frame_id_to_lineno = info
+
+                if trace_obj is not None:
+                    for frame_id, frame, method_name, filename_in_utf8, lineno in iter_visible_frames_info(
+                            dbg, trace_obj.tb_frame, frame_id_to_lineno):
+ 
+                        if sys.version_info < (3, 5,):
```

Comment:
Why get the line text only when < 3.5?

Reply:
`traceback.format_list` for `>=3.5` gets the source is line_text is `None`, we don't really need to provide that. Since we are using `linecache`, this check is probably not needed.